### PR TITLE
DLS-4257 Library upgrades for amls-frontend

### PR DESCRIPTION
**Pull request by rajanimohan22**
Library upgrades for amls-frontend

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [ ] Build passing.
- [ ] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [ ] Requires acceptance test run.
- [ ] Appropriate labels added.
- [ ] RELEASE NOTES ADDED.

### CODE REVIEW

File: app/config/AmlsHeaderCarrierForPartialsConverter.scala [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package config
+/*package config
```
Comment (calinursaciuc) [inline app/config/AmlsHeaderCarrierForPartialsConverter.scala:17]: Shall we delete this one?

File: app/services/UpdateMongoCacheService.scala [hunk 2]
```diff
@@ -76,14 +77,14 @@ class UpdateMongoCacheService @Inject()(http: HttpClient, val cacheConnector: Da
...
 
-    http.GET[UpdateMongoCacheResponse](requestUrl)
+    http.GET[UpdateMongoCacheResponse](requestUrl, Seq.empty, Seq.empty)
       .map { r =>
         import utils.Strings._
         println(Json.prettyPrint(Json.toJson(r)) in Console.YELLOW)
         Some(r.copy(dataImport = Some(DataImport(fileName)))) }
       .recover {
-        case e => throw e
         case _: NotFoundException => None
```
Comment (calinursaciuc) [inline app/services/UpdateMongoCacheService.scala:86]: I wonder if this also needs to be UpstreamError and maybe removing empty headers and query

File: app/utils/AuthAction.scala [hunk 3]
```diff
@@ -62,7 +62,10 @@ class DefaultAuthAction @Inject() (val authConnector: AuthConnector,
 
   override final protected def refine[A](request: Request[A]): Future[Either[Result, AuthorisedRequest[A]]] = {
 
-    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+    //implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request.headers, Some(request.session))
```
Comment (calinursaciuc) [inline app/utils/AuthAction.scala:65]: remove the comment

File: build.sbt [hunk 4]
```diff
@@ -35,13 +33,13 @@ lazy val scoverageSettings = {
...
-  .enablePlugins(Seq(play.sbt.PlayScala,SbtAutoBuildPlugin, SbtDistributablesPlugin, SbtGitVersioning, SbtArtifactory) ++ plugins : _*)
+  .enablePlugins(Seq(play.sbt.PlayScala, SbtDistributablesPlugin) ++ plugins : _*)
   .settings(majorVersion := 4)
   .settings(playSettings ++ scoverageSettings : _*)
   .settings(scalaSettings: _*)
   .settings(publishingSettings: _*)
   .settings(defaultSettings(): _*)
-  .settings(scalaVersion := "2.12.11")
+  .settings(scalaVersion := "2.12.13")
   .settings(routesImport += "models.notifications.ContactType._")
```
Comment (rambabu-posa) [inline build.sbt:43]: I think you can use this one: `coverageMinimumStmtTotal`

File: project/AppDependencies.scala [hunk 5]
```diff
@@ -5,16 +5,15 @@ private object AppDependencies {
...
 
-  private val playPartialsVersion = "6.11.0-play-26"
-  private val httpCachingClientVersion = "9.1.0-play-26"
+  private val playPartialsVersion = "8.1.0-play-26"
+  private val httpCachingClientVersion = "9.5.0-play-26"
   private val playAllowListFilterVersion = "3.4.0-play-26"
   private val validationVersion = "2.1.0"
   private val flexmarkVersion = "0.19.1"
   private val okHttpVersion = "3.9.1"
   private val jsonEncryptionVersion = "4.8.0-play-26"
```
Comment (rambabu-posa) [inline project/AppDependencies.scala:14]: Any issues with this version: `4.10.0-play-26`

File: build.sbt [hunk 6]
```diff
@@ -35,13 +33,13 @@ lazy val scoverageSettings = {
...
   .settings(routesImport += "utils.Binders._")
   .settings(
```
Comment (rambabu-posa) [inline build.sbt:45]: We can remove this here `esolvers += Resolver.jcenterRepo` (plugins.sbt have it)

File: build.sbt (hunk 4 above)
Comment (rambabu-posa) [inline build.sbt:43]: If we dont have any issues with `views` like too many warnings, we can remove it from silencer filter
 
`-P:silencer:pathFilters=views;routes;TestStorage`

File: project/AppDependencies.scala [hunk 7]
```diff
@@ -24,11 +23,9 @@ private object AppDependencies {
...
     "uk.gov.hmrc" %% "json-encryption" % jsonEncryptionVersion,
     "uk.gov.hmrc" %% "simple-reactivemongo" % playReactivemongoVersion,
-    "uk.gov.hmrc" %% "auth-client" % authVersion,
-    "uk.gov.hmrc" %% "play-ui" % "8.15.0-play-26",
-    "uk.gov.hmrc" %% "bootstrap-frontend-play-26" % "2.25.0",
-    "uk.gov.hmrc" %% "govuk-template" % "5.58.0-play-26",
-    "uk.gov.hmrc" %% "http-verbs" % "11.7.0",
+    "uk.gov.hmrc" %% "play-ui" % "9.4.0-play-26",
+    "uk.gov.hmrc" %% "bootstrap-frontend-play-26" % "5.3.0",
+    "uk.gov.hmrc" %% "govuk-template" % "5.66.0-play-26",
```
Comment (rambabu-posa) [inline project/AppDependencies.scala:28]: I think we can use this version if we don't see any compilation issues
`"silencer-plugin" % "1.7.5" `

File: build.sbt (hunk 4 above)
Comment (rajanimohan22) [reply build.sbt:43]: Done

File: project/AppDependencies.scala (hunk 5 above)
Comment (rajanimohan22) [reply project/AppDependencies.scala:14]: Done

File: build.sbt (hunk 6 above)
Comment (rajanimohan22) [reply build.sbt:46]: Done

File: build.sbt (hunk 4 above)
Comment (rajanimohan22) [reply build.sbt:43]: Not needed as we do not have any warnings from views.

File: project/AppDependencies.scala (hunk 7 above)
Comment (rajanimohan22) [reply project/AppDependencies.scala:28]: Done

**Pull request merged**